### PR TITLE
  Improve code action node validator

### LINKED ISSUE #36545: Create variable code action suggests where it is not expected
**Description:**
Code action `create variable` suggests for this following scenario, where it is not expected. (Same with the `ignore return value` code action). 

https://user-images.githubusercontent.com/46120162/173785062-7c203c99-468d-40e5-b8aa-32a17d31e86a.mov

**Steps to reproduce:**
```
import ballerina/http;

public function main() {
    http:Response json
}
```

**Affected Versions:**
2201.1.0

### PR DESCRIPTION
**Pull request by malinthar**
## Purpose
$subject to improve the validation logic in the code action node validator to look for syntax errors in the TypeTestNode.

Fixes #36545 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/test/resources/codeaction/create-variable/config/createVariableNegative1.json [hunk 1]
```diff
@@ -1,8 +1,12 @@
 {
   "position": {
     "line": 2,
-    "character": 16
+    "character":16
```
Comment (dulajdilshan) [inline language-server/modules/langserver-core/src/test/resources/codeaction/create-variable/config/createVariableNegative1.json:4]: unnecessary change?

File: language-server/modules/langserver-core/src/test/resources/codeaction/create-variable/config/createVariableNegative2.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+  "position": {
+    "line": 3,
+    "character":27
```
Comment (dulajdilshan) [inline language-server/modules/langserver-core/src/test/resources/codeaction/create-variable/config/createVariableNegative2.json:4]:
```suggestion
    "character": 27
```

**Review by dulajdilshan — COMMENTED** [review]
Minor suggestions. Rest is LGTM!

**Review by IMS94 — DISMISSED** [review]

**Review by dulajdilshan — APPROVED** [review]
LGTM

**Review by IMS94 — APPROVED** [review]

**Review by mohanvive — APPROVED** [review]
